### PR TITLE
Update NOTICE.txt to include link to certifi source

### PR DESCRIPTION
**Pull request by wangch079**
Connector service uses a transitive library [certifi](https://pypi.org/project/certifi/), which is under Mozilla Public License 2.0 (MPL 2.0). MPL 2.0 is in yellow list:
>Incorporation of **unmodified** source or binaries into Elastic products is permitted, provided that the product's NOTICE file links to a URL providing the MPL-covered source code

This PR adds the link to certifi source in NOTICE.txt

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] this PR has a thorough description
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: NOTICE.txt [hunk 1]
```diff
@@ -1,2 +1,4 @@
-connectors-py
+Elastic Python connectors
```
Comment (wangch079) [inline NOTICE.txt:1]: Not sure whether this is our official product name.

File: NOTICE.txt [hunk 2]
```diff
@@ -1,2 +1,4 @@
...
 Copyright 2022 Elasticsearch B.V.
```
Comment (wangch079) [inline NOTICE.txt:2]: Should this be
```suggestion
Copyright 2022-2023 Elasticsearch B.V.
```

File: NOTICE.txt (hunk 1 above)
Comment (artem-shelkovnikov) [reply NOTICE.txt:1]: This should be just the name of the repository, following our open-source guidelines (so `connectors-python`). I subtly remember this discussion with @gtback but don't remember where it was

File: NOTICE.txt (hunk 2 above)
Comment (artem-shelkovnikov) [reply NOTICE.txt:2]: Probably just 2023?

**Review by artem-shelkovnikov — COMMENTED** [review]
Looks good, left a couple small nits.

File: NOTICE.txt (hunk 1 above)
Comment (wangch079) [reply NOTICE.txt:1]: According to [the `NOTICE.txt` requirement for single-licence repo](https://github.com/elastic/open-source/blob/main/elastic-product-policy.md#for-repositories-holding-only-code-subject-to-the-elastic-license-egml-cpp), it should be the product name:
>{Product Name}
Copyright 2012-2018 Elasticsearch B.V.

One example: https://github.com/elastic/elasticsearch-ruby/blob/main/NOTICE

File: NOTICE.txt (hunk 2 above)
Comment (wangch079) [reply NOTICE.txt:2]: I didn't find any doc about this, but my understanding is it's a range from when the software was first developed to the current year. cc @elastic/open-source-wg

File: NOTICE.txt (hunk 1 above)
Comment (artem-shelkovnikov) [reply NOTICE.txt:1]: Ha didn't know, makes sense!

**Comment by wangch079** [comment]
>Have you checked this with our #open-source team?

@gtback confirmed [here](https://github.com/elastic/open-source/issues/382#issuecomment-1734446260):
>I confirm that the PR you linked to provides appropriate NOTICE of the Source URL. Please make sure that the `NOTICE.txt` file is included in all distributions of the package (PyPI, Docker images, etc.).

**Pull request merged**